ci/provisioning/post_provision_config_common_functions.sh

```diff
@@ -193,7 +193,8 @@ set_local_repo() {
     version=${version%%.*}
     if [ "$repo_server" = "artifactory" ] && [ -z "$(rpm_test_version)" ] &&
        { [[ ${CHANGE_TARGET:-$BRANCH_NAME} != weekly-testing* ]] ||
-         [[ ${CHANGE_TARGET:-$BRANCH_NAME} != provider-testing* ]]; }; then
+         [[ ${CHANGE_TARGET:-$BRANCH_NAME} != provider-testing* ]] ||
+         [[ ${CHANGE_TARGET:-$BRANCH_NAME} != soak-testing* ]]; }; then
```

Comment:
I'm starting to feel like we've made a mistake and all of this kind of weekly/provider/soak/etc. testing that is being done with already-existing RPMs should have been identified in some other way than this continual churn of adding more and more branch name patterns.  Heck, even if we'd have grouped them under a common branch _prefix/_ it would have scaled better than this is currently.  Maybe we need to bite the bullet and just do that.

Reply:
@phender Would it be horrible for continuity and whatnot if we were to do the above, basically resetting the test history of all of those branches back to 1 (albeit while not losing the current history)?

Reply:
> @phender Would it be horrible for continuity and whatnot if we were to do the above, basically resetting the test history of all of those branches back to 1 (albeit while not losing the current history)?

So are saying something like the current `provider-testing-tcp` branch would be replaced/renamed `rpm-testing/master/provider/tcp`?

Reply:
If so can whatever new branch name scheme we want to apply be handled by a git branch rename (where I'm optimistically hopeful that the build numbers are retained) - as opposed to needing to create new branches?  While new branches wouldn't be a big problem I would be concerned with having to keep the old branches open as to not loose the past test results.

Reply:
> So are saying something like the current `provider-testing-tcp` branch would be replaced/renamed `rpm-testing/master/provider/tcp`?

Yeah, something like that.  Then we can just look for a startswith('rpm-testing/') and never have to touch that code again.

> If so can whatever new branch name scheme we want to apply be handled by a git branch rename (where I'm optimistically hopeful that the build numbers are retained) - as opposed to needing to create new branches?

I'm not sure if you can rename branches remotely.  But even if you can, I am less optimistic that Jenkins will understand that it was a branch rename and maintain the history.  An experiment would prove or disprove that.

> While new branches wouldn't be a big problem I would be concerned with having to keep the old branches open as to not loose the past test results.

Yes, this would be true.  They would have to stay open until their history is no longer considered useful.  Assuming Jenkins can't handle a rename.

Reply:
It's also interesting to consider:

```
$ git branch -a | grep -e 'remotes\/origin\/[^\/]*-testing.*$'
  remotes/origin/ci-daos-stack-pipeline-lib-PR-329-weekly-testing-2.0
  remotes/origin/ci-daos-stack-pipeline-lib-PR-329-weekly-testing-2.2
  remotes/origin/provider-testing-tcp
  remotes/origin/provider-testing-tcp-2.2
  remotes/origin/provider-testing-ucx
  remotes/origin/provider-testing-ucx-2.2
  remotes/origin/release-testing-ucx
  remotes/origin/soak-testing
  remotes/origin/weekly-testing
  remotes/origin/weekly-testing-1.2
  remotes/origin/weekly-testing-1.x
  remotes/origin/weekly-testing-2.0
  remotes/origin/weekly-testing-2.2
  remotes/origin/weekly-testing-add-valgrind-stage
```

is (almost) a list entirely of what we are looking for.  There are a couple of exceptions, but I suspect those are abandoned branches that can be deleted.

It's particularly interesting considering that PR branch names are not allowed to use the root namespace any more and must be created under a per-user prefix so we shouldn't get any more exceptions.

So maybe we already have our _forever pattern_.  Although, not as nice as the hierarchy that @phender was suggesting.  Maybe this could be _plan B_.

Reply:
So would this suffice?
```
    if [ "$repo_server" = "artifactory" ] && [ -z "$(rpm_test_version)" ] &&
       { [[ ${CHANGE_TARGET:-$BRANCH_NAME} !=~ ^(weekly|provider|soak)-testing ]]; }; then
```

Or are you saying we want to be more generic:?
```
    if [ "$repo_server" = "artifactory" ] && [ -z "$(rpm_test_version)" ] &&
       { [[ ${CHANGE_TARGET:-$BRANCH_NAME} !=~ ^[a-z]+-testing ]]; }; then
```

Reply:
> So would this suffice?
> 
> ```
>     if [ "$repo_server" = "artifactory" ] && [ -z "$(rpm_test_version)" ] &&
>        [[ ! ${CHANGE_TARGET:-$BRANCH_NAME} =~ ^(weekly|provider|soak)-testing ]]; then
> ```

This still means editing this this file every time we add a new branch.
 
> Or are you saying we want to be more generic:?
> 
> ```
>     if [ "$repo_server" = "artifactory" ] && [ -z "$(rpm_test_version)" ] &&
>        [[ ! ${CHANGE_TARGET:-$BRANCH_NAME} =~ ^[a-z]+-testing ]]; then
> ```

Something like that, yeah.  I just want reduce the churn on this chunk of code so that we can have a convention that covers all future branches and not have to keep editing this whitelist.

Reply:
Shall we do something about this, either by doing something along the lines of the previous discussions about it or doing something completely different but that should be able to apply generically to new branches of this sort being added in the future without having to keep coming back to modify this code?